### PR TITLE
Add missing match_only_text field mapping

### LINKED ISSUE #1548: Missing match_only_text field mapping.
## 🐛 Missing type

The `match_only_text` field mapping is missing. See [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/text.html#match-only-text-field-type) and  https://github.com/elastic/elasticsearch/pull/66172

Found in https://github.com/elastic/elasticsearch-java/issues/208

### PR DESCRIPTION
**Pull request by philkra**
closes #1548

### CODE REVIEW

File: specification/_types/mapping/Property.ts [hunk 1]
```diff
@@ -65,47 +65,47 @@ export type Property =
...
+  alias,
+  join,
+  rank_feature,
+  rank_features,
+  flattened,
+  shape,
+  histogram,
+  constant_keyword,
+  aggregate_metric_double,
+  dense_vector
```
Comment (swallez) [inline specification/_types/mapping/Property.ts:108]: We have to add `match_only_text` here.

File: specification/_types/mapping/core.ts [hunk 2]
```diff
@@ -241,11 +242,33 @@ export class SearchAsYouTypeProperty extends CorePropertyBase {
...
+   * field for search and a multi-field for sorting and aggregations, or the same string value analyzed by different analyzers.
+   * @doc_url https://www.elastic.co/guide/en/elasticsearch/reference/current/multi-fields.html
+   */
+  fields?: Dictionary<PropertyName, Property>
+  /**
+   * Metadata about the field.
+   * @doc_id mapping-meta-field
+   */
+  meta?: Dictionary<string, string>
+}
```
Comment (swallez) [inline specification/_types/mapping/core.ts:265]: `copy_to` is also applicable to this field mapping:

```suggestion
  /**
   * Allows you to copy the values of multiple fields into a group
   * field, which can then be queried as a single field.
   */
  copy_to?: Fields
}
```

Comment (swallez) [inline specification/_types/mapping/core.ts:265]: `copy_to` is also applicable on this type:

```suggestion
  copy_to?: Fields
}
```

**Review by swallez — CHANGES_REQUESTED** [review]

File: specification/_types/mapping/Property.ts (hunk 1 above)
Comment (philkra) [reply specification/_types/mapping/Property.ts]: addressed in https://github.com/elastic/elasticsearch-specification/pull/1549/commits/16215d6c446ac5726a4a23876b4bfdecf018d4cf

**Pull request merged**